### PR TITLE
Fix some NanoUI checks

### PR DESCRIPTION
**Pull request by neersighted**
Canisters require physical access and nothing else.
NanoUI checks Brain Damage by default.
Fixes a NanoUI assignment derp.
Uses two-line ifs in Nano machines.

:cl: neersighted
tweak: NanoUI becomes... difficult to operate with brain damage. Who would have guessed?
tweak: Telekinesis enables you to use NanoUI at a distance.
fix: Canisters require physical proximity to actuate...
fix: NanoUI no longer leaks memory on click.
/:cl:

### CODE REVIEW

**Comment by neersighted** [comment]
It seems that the brain damage checks are a bit controversial... One one hand, we can make it harder but not impossible to use. I implemented a probability check based on the level of brain damage. This disables the UI `brainloss`% of the time.

On the other hand, some people prefer a flat check (`>= 60`) and a the UI to be blocked entirely (presumably with a message).

Can I get some opinions here?

File: code/modules/nano/states/base.dm [hunk 1]
```diff
@@ -40,19 +40,27 @@
   *
   * return NANO_state The state of the UI.
  **/
-/mob/proc/shared_nano_interaction()
+/mob/proc/shared_nano_interaction(atom/movable/src_object)
 	if (!client || stat) // Close NanoUIs if mindless or dead/unconcious.
 		return NANO_CLOSE
-	else if (restrained() || lying || stat || stunned || weakened) // Update NanoUIs if incapicitated but concious.
+	// Update NanoUIs if incapicitated but concious.
+	else if (restrained() || incapacitated() || lying || stunned || weakened)
```
Comment (phil235) [inline code/modules/nano/states/base.dm]: /mob/living/incapacitated() already checks for restrained(), stunned and weakened.

**Comment by phil235** [comment]
I prefer your implementation of the brain damage check. It's less frustrating for the player.

**Comment by Iamgoofball** [comment]
>neersighted commented 23 hours ago

**Pull request merged**